BBallStatsTrack/src/com/example/bballstatstrack/models/utils/JSONDeserializer.java

```diff
@@ -0,0 +1,431 @@
+
+import android.util.JsonReader;
+import android.util.SparseArray;
+
+public class JSONDeserializer
+{
+    private static Team mHomeTeam = null;
+
+    private static Team mAwayTeam = null;
+
+    public static Game readGame( JsonReader reader ) throws IOException
+    {
+        UUID id = null;
+        long longDate = Long.MIN_VALUE;
+        GameLog gameLog = null;
+        reader.beginObject();
+        while( reader.hasNext() )
+        {
+            String name = reader.nextName();
+            if( name.equals( GameStats.ID.toString() ) )
+            {
+                id = UUID.fromString( reader.nextString() );
+            }
+            else if( name.equals( GameStats.DATE.toString() ) )
+            {
+                longDate = reader.nextLong();
+            }
+            else if( name.equals( GameStats.HOME_TEAM.toString() ) )
+            {
+                mHomeTeam = readTeam( reader );
+            }
+            else if( name.equals( GameStats.AWAY_TEAM.toString() ) )
+            {
+                mAwayTeam = readTeam( reader );
+            }
+            else if( name.equals( GameStats.GAME_LOG.toString() ) )
+            {
+                gameLog = readGameLog( reader );
+            }
+        }
+        reader.endObject();
+        return new Game( id, longDate, mHomeTeam, mAwayTeam, gameLog );
+    }
+
+    private static Team getOtherTeam( Team team )
+    {
+        if( team.equals( mHomeTeam ) )
+        {
+            return mAwayTeam;
+        }
+        else if( team.equals( mAwayTeam ) )
+        {
+            return mHomeTeam;
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+    private static Player getPlayerFromTeam( Team team, int playerNumber )
+    {
+        return team.getPlayers().get( playerNumber );
+    }
+
+    private static Player getPlayerNumberFromOtherTeam( Team team, int number )
+    {
+        Team otherTeam = getOtherTeam( team );
+        return otherTeam.getPlayers().get( number );
+    }
+
+    private static Team getTeamFromUUID( UUID teamID )
+    {
+        if( teamID.equals( mHomeTeam.getID() ) )
+        {
+            return mHomeTeam;
+        }
+        else if( teamID.equals( mAwayTeam.getID() ) )
+        {
+            return mAwayTeam;
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+    private static GameEvent readEvent( JsonReader reader ) throws IOException
+    {
+        EventType type = null;
+        int playerNumber = Integer.MIN_VALUE;
+        UUID teamID = null;
+        int time = Integer.MIN_VALUE;
+        GameEvent appended = null;
+        GameEvent thisEvent = null;
+        reader.beginObject();
+        if( reader.nextName().equals( GameEvent.EVENT_TYPE ) )
+        {
+            type = EventType.valueOf( reader.nextString() );
+        }
+        if( reader.nextName().equals( GameEvent.PLAYER_NUMBER ) )
```

Comment:
Reworked this as well. I removed the  if else conditions and instead just consumed all names since it has an equivalent in the JSONSerializer.